docs/prover/techniques/index.md

```diff
@@ -1,31 +1,121 @@
-Techniques Used by Certora Prover
-=================================
+Techniques Used by the Certora Prover
+=====================================
 
-In this chapter we describe techniques used by the Certora Prover whose understanding can be relevant for an expert-level usage of the Prover.
+In this chapter, we describe some of the techniques used inside the Certora
+Prover. While this knowledge is not essential for using the Prover, it can
+sometimes be helpful when the Prover does not behave as expected, for instance
+in case of a timeout.
 
 (control-flow-splitting)=
-# Control Flow Splitting
+# Control flow splitting
 
-There is a brief explanation of control flow splitting in the 
+
+```{note}
+In addition to the text-form documentation below,  there is a brief explanation 
+of control flow splitting in the 
 [webinar on timeouts](https://www.youtube.com/watch?v=mntP0_EN-ZQ).
+```
+
+
+Control flow splitting (or short "splitting") is one of the techniques that the
+Certora Prover employs to speed up solving. In the remainder of this section, we
+will give an overview of how the technique works. This background should be
+helpful when using the settings described [here](control-flow-splitting-options)
+to prevent Prover timeouts.
+
+Splitting is best illustrated using the {term}`control flow graph` (CFG) of a given
+CVL rule.
+
+A single splitting step proceeds as follows:
```

Comment:
I think there is a high-level idea that might get lost in the details.  Essentially whenever we find a branch (e.g. an `if` statement or loop guard), we sometimes look for counterexamples in each branch separately.  We could also include a small example, something like:

````
For example, consider a rule with an `if` statement:
```cvl
rule example {
  ...
  if (owner == spender) {
    assert balance_after == balance_before;
  } else {
    assert balance_after == balance_before + amount;
  }
}
```
To simplify the search for a counterexample, the Prover may internally split this single rule into two rules:
```cvl
rule example_split_1 {
  ...
  require owner == spender;
  assert balance_after == balance_before;
}

rule example_split_2 {
  ...
  require owner != spender;
  assert balance_after == balance_before + amount;
}
```

A counterexample for either of the split rules will also be a counterexample for the original rule, and any counterexample for the original rule must violate one of the two split rules, so this splitting doesn't change the meaning of the rule.  However, in some cases the split rule is easier for the Prover to reason about.

Reply:
Right, making this into an intro section -- should be nice!